tests/UnsafeInterfacesTest.php

```diff
@@ -48,4 +58,15 @@ public function testAlwaysValidChild() {
     $xhp = <html>{$x}<body /></html>;
     $this->assertEquals('<html>foo<body></body></html>', $xhp->toString());
   }
+
+  public function testUnsafeAttribute(): void {
+    // without using XHPUnsafeAttributeValue, this will be escaped in an undesirable way
+    $xhp = <script onload="foo && bar" />;
+    $this->assertEquals('<script onload="foo &amp;&amp; bar"></script>', $xhp->toString());
+
+    // skip escaping so that the JS works
+    $escaped = new ExampleUnsafeAttribute('foo && bar');
+    $xhp = <script onload={$escaped} />;
+    $this->assertEquals('<script onload="foo && bar"></script>', $xhp->toString());
+  }
```

Comment:
I don't understand this example - this works fine:

```
<html>
<div data-foo="foo &amp; bar" />
<script>
alert(document.querySelector('div').dataset.foo);
</script>
<div onclick="javascript: alert('foo &amp; bar');">Click me</div>
```

Reply:
You're right that this also does work at least in chrome, safari, and firefox, even though it looks like a problem to me.

```
<html>
<--! alert does fire on click -->
<div onclick="true &amp;&amp; alert('hello');">Click me</div>

<--! alert does not fire on click -->
<div onclick="false &amp;&amp; alert('hello');">Click me</div>
</html>
```

Maybe I'm being overly paranoid in seeing this as a problem?

Reply:
This behavior is required (https://www.w3.org/TR/html51/syntax.html#elements-attributes), and is fine for &&

That said, for migration, this might be a concern (and better example for the unit test):

```
<html>
<div />
<div onclick="document.querySelector('div').innerHTML='foo &amp; bar'">click me</div>
```

The main downside of using something like this is that *if* you have a single-quoted onclick handler that uses double quotes, using UnsafeAttribute will break the runtime behavior